radioactivedecay/nuclide.py

```diff
@@ -0,0 +1,651 @@
+            Z = vary_input
+            name = self.build_nuclide_string(Z, A)
+        else:
+            raise TypeError(
+                "Invalid input types, expected int"
+            )
+        
+        self.nuclide_name = name
+        self.Z_val = Z
+        self.A_val = A
+    
+    def build_nuclide_string(
+        self,
+        Z: int,
+        A: int = 0,
+        meta_state: str = ""
+        ) -> str:
+                    
+        if Z not in self.Z_dict.keys():
+            raise ValueError(
+                str(Z) + ' is not a valid atomic number'
+            )
+            
+        return_string = self.Z_dict[Z]
+        if A != 0:
+            return_string += "-" + str(A) + meta_state
+            
+        return return_string
+    
+    def __repr__(self) -> str:
+        if self.A == 0:
+            rep =  "Element: " + self.nuclide_name
+        else:
+            rep = "Nuclide: " + self.nuclide_name + ", decay dataset: " + self.data.dataset
+            
+        return rep
+    
+    @property
+    def name(self) -> str:
+        """
+        Returns the element or nuclide name of the object.
+
+        Examples
+        --------
+        >>> rd.Nuclide("Fe-56").name
+        Fe-56
+        >>> rd.Nuclide("Fe56").name
+        Fe-56
+        >>> rd.Nuclide(26, 56).name
+        Fe-56
+        >>> rd.Nuclide("Fe").name
+        Fe
+        >>> rd.Nuclide(26).name
+        Fe
+
+        """
+        return self.nuclide_name
+    
+    @property
+    def Z(self) -> int:
+        """
+        Returns the atomic number of the element or nuclide.
+
+        Examples
+        --------
+        >>> rd.Nuclide("Fe-56").Z
+        26
+        >>> rd.Nuclide("Fe56").Z
+        26
+        >>> rd.Nuclide(26, 56).Z
+        26
+        >>> rd.Nuclide("Fe").Z
+        26
+        >>> rd.Nuclide(26).Z
+        26
+        """
+        
+        return self.Z_val
+    
+    @property
+    def A(self) -> int:
+        """
+        Returns the atomic mass of the nuclide, and 0 for elements.
+
+        Examples
+        --------
+        >>> rd.Nuclide("Fe-56").A
+        56
+        >>> rd.Nuclide("Fe56").A
+        56
+        >>> rd.Nuclide(26, 56).A
+        56
+        >>> rd.Nuclide("Fe").A
+        0
+        >>> rd.Nuclide(26).A
+        0
+        """
+        
+        return self.A_val
+    
+    @property
+    def atomic_mass(self) -> int:
+        """
+        Returns the atomic mass of the nuclide, if contained in the dataset.
+        
+        Raises
+        ------
+        ValueError
+            If the nuclide is not contained in the decay
+            dataset.
+
+        Examples
+        --------
+        >>> rd.Nuclide("Fe-56").atomic_mass
+        56
+        >>> rd.Nuclide("Fe56").atomic_mass
+        56
+        >>> rd.Nuclide(26, 56).atomic_mass
+        56
+        """
+        
+        if self.nuclide_name not in self.data.radionuclides:
+            raise ValueError(
```

Comment:
I think this shouldn't need to be checked here, as we should be checking the nuclide exists in the dataset upon instantiation of the object.